daemon/containerd/image_manifest.go

```diff
@@ -78,21 +128,40 @@ func (im *ImageManifest) Metadata() containerdimages.Image {
 	return md
 }
 
-// IsPseudoImage returns false if the manifest has no layers or any of its layers is a known image layer.
-// Some manifests use the image media type for compatibility, even if they are not a real image.
-func (im *ImageManifest) IsPseudoImage(ctx context.Context) (bool, error) {
-	desc := im.Target()
-
+func (im *ImageManifest) IsAttestation() bool {
 	// Quick check for buildkit attestation manifests
 	// https://github.com/moby/buildkit/blob/v0.11.4/docs/attestations/attestation-storage.md
 	// This would have also been caught by the layer check below, but it requires
 	// an additional content read and deserialization of Manifest.
-	if _, has := desc.Annotations[attestation.DockerAnnotationReferenceType]; has {
+	if _, has := im.Target().Annotations[attestation.DockerAnnotationReferenceType]; has {
+		return true
+	}
+	return false
+}
+
+// IsPseudoImage returns false when any of the below is true:
+// - The manifest has no layers
+// - None of its layers is a known image layer.
+// - The manifest has unknown/unknown platform.
+//
+// Some manifests use the image media type for compatibility, even if they are not a real image.
+func (im *ImageManifest) IsPseudoImage(ctx context.Context) (bool, error) {
```

Comment:
(probably not actionable for this PR, but thoughts while reading this code) Is the intent that this continually grows over time?  How do we handle non-images like WASI that _are_ still images depending on the runtime? :thinking:
(`IsNotStandardLayeredContainerImage` is a mouthful, but I think that's so far the intent I see here :sweat_smile:)

Reply:
IIRC, WASI images still have layers with `application/vnd.docker.image.rootfs.diff.tar*` media type, so they should pass this check.

In general, I'd like to avoid growing this function more, unless we need to explicitly filter out something that really pretends to be an image, but is not.

In fact, the "non standard layer" check should be enough for this function, but since it requires reading the manifest from the content store, it also has "fast paths" that are able to exclude known "pseudo" images without actually parsing the manifest json.

So I guess that `IsNotStandardLayeredContainerImage` makes sense looking at the implementation, I consider the "non standard layer" to be an implementation detail.
I think what we really want to know here is that if the image is _runnable_. So perhaps `IsRunnableImage` would make more sense here?